linera-views/src/key_value_store_view.rs

```diff
@@ -135,6 +153,12 @@ where
             }
             self.stored_hash = hash;
         }
+        if self.stored_total_size != self.total_size || self.was_cleared {
```

Comment:
No need to test `self.was_cleared`

Reply:
Ah.
Most of the time you are right and I am wrong. But in that case, I am right and it took me one hour debugging to find out that the `self.was_cleared` was needed.

So, suppose that you do a clearing, but the obtained size is the same as the stored size. Then the size is first removed by the original DeletePrefix and then the new size is not written which means that when we run again, we have a size of 0 which is incorrect.

Luckily, that rare bug was caught by the random checks.

Reply:
Gotcha. Need to add a comment then :)